.circleci/scripts/bundle-stats-commit.sh

```diff
@@ -40,41 +40,45 @@ git config --global user.name "MetaMask Bot"
 
 git clone git@github.com:MetaMask/extension_bundlesize_stats.git temp
 
+BUNDLE_SIZE_FILE="test-artifacts/chrome/bundle_size_stats.json"
+STATS_FILE="temp/stats/bundle_size_data.json"
+TEMP_FILE="temp/stats/bundle_size_data.temp.json"
+
+# Ensure the JSON file exists
+if [[ ! -f "$STATS_FILE" ]]; then
+    echo "{}" > "$STATS_FILE"
+fi
+
+# Validate JSON files before modification
+jq . "$STATS_FILE" > /dev/null || { echo "Error: Existing stats JSON is invalid"; exit 1; }
+jq . "$BUNDLE_SIZE_FILE" > /dev/null || { echo "Error: New bundle size JSON is invalid"; exit 1; }
+
+# Check if the SHA already exists in the stats file
+if jq -e "has(\"$CIRCLE_SHA1\")" "$STATS_FILE" > /dev/null; then
+    echo "SHA $CIRCLE_SHA1 already exists in stats file. No new commit needed."
+    exit 0
+fi
+
+# Append new bundle size data correctly using jq
+jq --arg sha "$CIRCLE_SHA1" --argjson data "$(cat "$BUNDLE_SIZE_FILE")" \
+   '. + {($sha): $data}' "$STATS_FILE" > "$TEMP_FILE"
+
+# Overwrite the original JSON file with the corrected version
+mv "$TEMP_FILE" "$STATS_FILE"
+
 {
     echo " '${CIRCLE_SHA1}': ";
-    cat test-artifacts/chrome/bundle_size_stats.json;
+    cat "$BUNDLE_SIZE_FILE";
     echo ", ";
 } >> temp/stats/bundle_size_data.temp.js
 
 cp temp/stats/bundle_size_data.temp.js temp/stats/bundle_size_data.js
-
 echo " }" >> temp/stats/bundle_size_data.js
 
-if [ -f temp/stats/bundle_size_data.json ]; then
-  # copy bundle_size_data.json in bundle_size_data.temp.json without last 2 lines
-  head -$(($(wc -l < temp/stats/bundle_size_data.json) - 2)) temp/stats/bundle_size_data.json > bundle_size_stats.temp.json
-
-  {
-    echo "},";
-    echo "\"$CIRCLE_SHA1\":";
-    cat test-artifacts/chrome/bundle_size_stats.json;
-    echo "}";
-  } >> bundle_size_stats.temp.json
-else
-  {
-    echo "{";
-    echo "\"$CIRCLE_SHA1\":";
-    cat test-artifacts/chrome/bundle_size_stats.json;
-    echo "}";
-  } > bundle_size_stats.temp.json
-fi
-
-jq . bundle_size_stats.temp.json > temp/stats/bundle_size_data.json
-rm bundle_size_stats.temp.json
-
 cd temp
 
-git add .
+# Only add the JSON file
+git add stats/bundle_size_data.json
```

Comment:
mmm I'm a bit unsure about this change, because at the moment, we are also using the `bundle-size-data.js` file

https://github.com/MetaMask/extension_bundlesize_stats/tree/main/stats

in the visualization tool, so not uploading it will cause that to be updated, no?

https://github.com/MetaMask/extension_bundlesize_stats/blob/main/background.html#L136

Reply:
I think Howard means we don't need the file `bundle_size_data.js` as it should have same data with  `bundle_size_data.json` , so we can only refer to `bundle_size_data.json`.
 But yeah what you pointed out makes sense, maybe we should first go into [extension_bundlesize_stats](https://github.com/MetaMask/extension_bundlesize_stats) to remove file `bundle_size_data.js` and `bundle-size-data-temp.js`, and also change the line you linked to refer to `bundle_size_data.json`.
 Converting this PR to draft, and tag @davidmurdoch for clarifications.

Reply:
@seaona @chloeYue I made the changes to [extension_bundlesize_stats](https://github.com/MetaMask/extension_bundlesize_stats), and this PR should be good to go now

Reply:
Thanks a lot @HowardBraham

Reply:
awesome , thank you so much both!